server-actions/user-actions.ts

```diff
@@ -21,25 +21,62 @@ export async function createUser(email: string) {
   });
 }
 
-export async function updateUserRewards(email: string, room_cost: number) {
-  try {
-    const pointsToAdd = room_cost * DEFAULT_REWARDS_MULTIPLIER;
+export async function getUserRewards(email: string) {
+  const user = await prisma.user.findUnique({
+    where: { email },
+    select: { rewardPoints: true },
+  });
 
-    const updatedRewards = await prisma.user.update({
-      where: {
-        email: email,
-      },
-      data: {
-        rewardPoints: {
-          increment: pointsToAdd,
-        },
-      },
-    });
+  if (!user) {
+    throw new Error("User not found");
+  }
+
+  return user.rewardPoints;
+}
 
-    console.log("Updated rewards:", updatedRewards);
-    return updatedRewards;
-  } catch (error: any) {
-    console.error("Error updating rewards:", error.message);
-    throw error;
+export async function updateUserRewards(email: string, payment: number) {
+  const user = await prisma.user.findUnique({
+    where: { email },
+    select: { rewardPoints: true },
+  });
+
+  if (!user) {
+    throw new Error("User not found");
   }
+  const oldPoints = user.rewardPoints;
+
+  const rewardPoints = oldPoints ?? 0 + payment * DEFAULT_REWARDS_MULTIPLIER;
```

Comment:
```suggestion
  const rewardPoints = (oldPoints ?? 0) + payment * DEFAULT_REWARDS_MULTIPLIER;
```

Creating a reservation when the user had 0 reward points did not update the reward points.
This should fix it.